src/pkg/services/m365/api/events.go

```diff
@@ -208,63 +219,223 @@ func (c Events) GetItem(
 		}
 	)
 
-	event, err = c.Stable.
+	// Beta endpoint helps us fetch the event exceptions, but since we
+	// don't use the beta SDK, the exceptionOccurrences and
+	// cancelledOccurrences end up in AdditionalData
+	// https://learn.microsoft.com/en-us/graph/api/resources/event?view=graph-rest-beta#properties
+	rawURL := fmt.Sprintf(eventExceptionsBetaURLTemplate, userID, itemID)
+	builder := users.NewItemEventsEventItemRequestBuilder(rawURL, c.Stable.Adapter())
+
+	event, err = builder.Get(ctx, config)
+	if err != nil {
+		return nil, nil, graph.Stack(ctx, err)
+	}
+
+	// Adding checks to ensure that the data is in the format that we expect M365 to return
+	cancelledOccurrences := event.GetAdditionalData()["cancelledOccurrences"]
+	if cancelledOccurrences != nil {
```

Comment:
wouldn't mind seeing these blocks broken out into separate functions, similar to what was done in events_restore.